WireAPI/Sources/WireAPI/APIs/UserPropertiesAPI/UserPropertiesAPIV0.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class UserPropertiesAPIV0: UserPropertiesAPI, VersionedAPI {
+
+    let httpClient: HTTPClient
+
+    init(httpClient: HTTPClient) {
+        self.httpClient = httpClient
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+
+    // MARK: - Fetch user property
+
+    func getProperty(forKey key: UserProperty.Key) async throws -> UserProperty {
+        let request = HTTPRequest(
+            path: "\(pathPrefix)/properties/\(key.rawValue)",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        switch key {
+        case .wireReceiptMode:
+            return try getProperty(response: response, payload: ReceiptModeResponseV0.self)
+        case .wireTypingIndicatorMode:
+            return try getProperty(response: response, payload: TypeIndicatorModeResponseV0.self)
+        case .labels:
+            return try getProperty(response: response, payload: LabelsResponseV0.self)
+        }
+    }
+
+    func getProperty<Payload: UserPropertiesResponseAPIV0>(
+        response: HTTPResponse,
+        payload: Payload.Type
+    ) throws -> UserProperty where Payload.APIModel == UserProperty {
```

Comment:
I suggest to make the name a bit more precise:

```suggestion
    func parseResponse<Payload: UserPropertiesResponseAPIV0>(
        _ response: HTTPResponse,
        forPayloadType type: Payload.Type
    ) throws -> UserProperty where Payload.APIModel == UserProperty {
```

Reply:
better 👍  thx